src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs

```diff
@@ -373,5 +390,10 @@ static string GetTargetAssembly (ITaskItem assemblyName)
 				jars.Add (f);
 			}
 		}
+
+		bool ShouldSkipAssembly (string assemblyFileName)
+		{
+			return AssembliesToSkipCases != null && AssembliesToSkipCases.Contains (assemblyFileName, StringComparer.OrdinalIgnoreCase);
```

Comment:
This is an O(n) search against an array with 10s of entries (`@(_AndroidAssemblySkipCases)`) which will be performed at least once for every assembly in the app.

Should we instead use a `HashSet<string>`, e.g. in `Execute()`:

```csharp
HashSet<string> AssembliesToSkip;

public override bool Execute ()
{
    AssembliesToSkip = new HashSet<string>(AssembliesToSkipCases ?? new string[0]);
    // ...
}
```

Reply:
👍 looks like `HashSet<string>` has an option to do case-insensitivity as well: https://stackoverflow.com/questions/16922435/make-hashsetstring-case-insensitive

I bet this will make the perf hit in `ResolveLibraryProjectImports` better.